src/species/__init__.py

```diff
@@ -21,42 +24,63 @@ def __init__(self, species, count):
         self.available_food = 0
         self.count_over_time = []
 
+    def calculate_development(self):
+        self.calculate_available_food()
+        # For it all to work properly, calculate decline for everything,
+        # apply the new values and then do a second round to add the incline
+
+    def save_for_graph(self):
+        self.count_over_time.append(self.count)
+
     def apply_development(self):
         self.count += self.development
-        self.count_over_time.append(self.count)
-        if self.count < 1:
-            self.count = 0
 
-    def calculate_development(self):
-        self.calculate_available_food()
-        self.development = self.calculate_incline() - self.calculate_decline()
+        if self.count < 2:
+            self.go_extinct()
+
+    def go_extinct(self):
+        self.count = 0
 
     def calculate_available_food(self):
-        if len(self.species.food) == 0 or self.count <= 0:
-            self.available_food = 1
-            return
+        if len(self.species.prey) == 0:
+            self.calculate_available_food_herbivores()
+        else:
+            self.calculate_available_food_carnivores()
+
+    def calculate_available_food_herbivores(self):
+        self.available_food = 1
+
+    def calculate_available_food_carnivores(self):
+        # TODO: Get all the possible prey.count and calculate how many of them could be consumed
+        # Do this for all predators in the end share all the values and if 2 or more species are competing,
+        # let them fight for overlapping prey. Then sum up all they've got, don't let them kill what they can't eat and
+        # give back the food amount
+
         food = self.available_food_count()
         if food > 0:
             self.available_food = food / self.count * 0.01
-            self.available_food = max(0, min(self.available_food, 1))
+            self.available_food = min(self.available_food, 1)
         else:
             self.available_food = 0
 
     def available_food_count(self):
         count = 0
         for population in self.tile.populations:
-            if population.species in self.species.food:
+            if population.species in self.species.prey:
                 count += population.count
         return count
 
     def calculate_incline(self):
-        return self.count * self.species.birth_rate * self.available_food
+        # TODO: Available food will belong to decline as it is not boosting growth but rather boosting decline when not available
+        self.development = self.count * self.species.birth_rate * self.available_food
 
     def calculate_decline(self):
+        if self.available_food <= 0:
+            return self.count
```

Comment:
I won't kill the whole species there is a part that would survive we should make a const with around 0.7 and multiply this with the count first.